addons/binding/org.openhab.binding.phc/src/main/java/org/openhab/binding/phc/handler/PHCHandler.java

```diff
@@ -0,0 +1,172 @@
+import org.eclipse.smarthome.core.library.types.UpDownType;
+import org.eclipse.smarthome.core.thing.Bridge;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandler;
+import org.eclipse.smarthome.core.thing.binding.ThingHandler;
+import org.eclipse.smarthome.core.types.Command;
+import org.eclipse.smarthome.core.types.State;
+import org.openhab.binding.phc.PHCBindingConstants;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link PHCHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Jonas Hohaus - Initial contribution
+ *
+ */
+public class PHCHandler extends BaseThingHandler {
+
+    private Logger logger = LoggerFactory.getLogger(PHCHandler.class);
+
+    private String moduleAddress; // like DIP switches
+    private short[] upDownTimes = new short[4];
+    private Map<String, State> channelState = new HashMap<String, State>();
+    private PHCBridgeHandler bridgeHandler;
+
+    public PHCHandler(Thing thing) {
+        super(thing);
+
+    }
+
+    @Override
+    public void initialize() {
+        try {
+            moduleAddress = (String) getConfig().get(PHCBindingConstants.ADDRESS);
+
+            if (getPHCBridgeHandler() == null) {
+                return;
+            }
+
+            getPHCBridgeHandler()
+                    .addModule(Byte.parseByte(getThing().getThingTypeUID().equals(PHCBindingConstants.THING_TYPE_EM)
+                            ? new StringBuilder(moduleAddress).reverse().toString()
+                            : ("010" + new StringBuilder(moduleAddress).reverse().toString()), 2)); // 010x = 0x4x for
+                                                                                                    // AM and JRM
+
+            if (getThing().getThingTypeUID().equals(PHCBindingConstants.THING_TYPE_JRM)) {
+                upDownTimes[0] = (short) (((BigDecimal) getConfig().get(PHCBindingConstants.UP_DOWN_TIME_1))
+                        .shortValue() * 10);
+                upDownTimes[1] = (short) (((BigDecimal) getConfig().get(PHCBindingConstants.UP_DOWN_TIME_2))
+                        .shortValue() * 10);
+                upDownTimes[2] = (short) (((BigDecimal) getConfig().get(PHCBindingConstants.UP_DOWN_TIME_3))
+                        .shortValue() * 10);
+                upDownTimes[3] = (short) (((BigDecimal) getConfig().get(PHCBindingConstants.UP_DOWN_TIME_4))
+                        .shortValue() * 10);
+            }
+
+            if (getBridge().getStatus() == ThingStatus.ONLINE) {
+                updateStatus(ThingStatus.ONLINE);
+            } else {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.BRIDGE_OFFLINE);
+            }
+
+        } catch (Exception e) {
```

Comment:
Do you really need to catch the Exception-class or are there more specific exceptions you could catch like the IOException or maybe only RuntimeException's (which is not that good either). The current code and also catching RuntimeException's might 'hide' programming errors like NullPointerExceptions occurring.

https://www.google.nl/search?q=java%20why%20not%20to%20catch%20exception